src/NuGet.Core/NuGet.PackageManagement/IDE/PackageRestoreManager.cs

```diff
@@ -377,6 +378,22 @@ private NuGetPackageManager GetNuGetPackageManager(string solutionDirectory)
 
             packageRestoreContext.Token.ThrowIfCancellationRequested();
 
+            IEnumerable<SourceRepository> enabledSources =
+                packageRestoreContext
```

Comment:
packageRestoreContext has the SourceRepositories. Can we use those?

Reply:
I think we can use `packageRestoreContext.SourceRepositories` because the collection is populated with sources that are enabled.

https://github.com/NuGet/NuGet.Client/blob/aca94657c6cf558c69bb3b5228856581c8c5b210/src/NuGet.Core/NuGet.Protocol/CachingSourceProvider.cs#L33-L36

Reply:
I had to add back this change, because some code paths `sourceRepositories` is set explicitly null so we can't use `packageRestoreContext.SourceRepositories` that case.

https://github.com/NuGet/NuGet.Client/blob/f52a83999070237288d7e1c8ec5934a65eda77ac/src/NuGet.Core/NuGet.PackageManagement/IDE/PackageRestoreManager.cs#L312

Reply:
I'd really love it if we can avoid adding *any* public methods in NuGetPackageManager. 

I tried to follow RestorePackageAsync into NuGetPackageManager and it uses the ISourceRepositoryProvider to get the providers if they are not set on the context. 

Maybe we can just set them?